docs/src/main/asciidoc/spring-data-jpa-guide.adoc

```diff
@@ -0,0 +1,598 @@
+INSERT INTO fruit(id, name, color) VALUES (2, 'Apple', 'Red');
+INSERT INTO fruit(id, name, color) VALUES (3, 'Banana', 'Yellow');
+INSERT INTO fruit(id, name, color) VALUES (4, 'Avocado', 'Green');
+----
+
+Hibernate ORM will execute these queries on application startup.
+
+== Define the repository
+
+It is now time to define the repository that will be used to access `Fruit`.
+In a typical Spring Data fashion create a repository like so:
+
+[source,java]
+----
+package org.acme.spring.data.jpa;
+
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.List;
+
+public interface FruitRepository extends CrudRepository<Fruit, Long> {
+
+    List<Fruit> findByColor(String color);
+}
+----
+
+The `FruitRepository` above extends Spring Data's `org.springframework.data.repository.CrudRepository` which means that all of the latter's methods are
+available to `FruitRepository`.
+Additionally `findByColor` is defined whose purpose is to return all Fruit entities that match the specified color.
+
+== Update the JAX-RS resource
+
+With the repository in place, the next order of business is to update the JAX-RS resource that will use the `FruitRepository`. Open `FruitResource` and change its contents to:
+
+[source,java]
+----
+package org.acme.spring.data.jpa;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import java.util.List;
+import java.util.Optional;
+
+@Path("/fruits")
+public class FruitResource {
+
+    private final FruitRepository fruitRepository;
+
+    public FruitResource(FruitRepository fruitRepository) {
+        this.fruitRepository = fruitRepository;
+    }
+
+    @GET
+    @Produces("application/json")
+    public Iterable<Fruit> findAll() {
+        return fruitRepository.findAll();
+    }
+
+
+    @DELETE
+    @Path("{id}")
+    public void delete(@PathParam("id") long id) {
+        fruitRepository.deleteById(id);
+    }
+
+    @POST
+    @Path("/name/{name}/color/{color}")
+    @Produces("application/json")
+    public Fruit create(@PathParam("name") String name, @PathParam("color") String color) {
+        return fruitRepository.save(new Fruit(name, color));
+    }
+
+    @PUT
+    @Path("/id/{id}/color/{color}")
+    @Produces("application/json")
+    public Fruit changeColor(@PathParam("id") Long id, @PathParam("color") String newColor) {
+        Optional<Fruit> optional = fruitRepository.findById(id);
+        if (optional.isPresent()) {
+            Fruit fruit = optional.get();
+            fruit.setColor(newColor);
+            return fruitRepository.save(fruit);
+        }
+
+        throw new IllegalArgumentException("No Fruit with id " + id + " exists");
+    }
+
+    @GET
+    @Path("/color/{color}")
+    @Produces("application/json")
+    public List<Fruit> all(@PathParam("color") String color) {
+        return fruitRepository.findByColor(color);
```

Comment:
So my comment was my more that you are not returning `all` fruits, you are searching by color. So I wonder why you name the method `all`.

Reply:
Right, my bad!